plutus-core/untyped-plutus-core/test/Flat/Spec.hs

```diff
@@ -85,6 +90,120 @@ test_canonicalByteString :: TestTree
 test_canonicalByteString =
   test_canonicalEncoding @BS.ByteString "flat encodes ByteStrings canonically" 1000
 
+{- Some tests that non-canonically encoded bytestrings decode correctly to strict
+bytestrings.  One strategy is to encode lazy bytestrings and decode the results
+to get strict bytestrings and then check that the result is the same as
+converting the original input into a strict bytestring, and we do this with
+arbitrary lazy bytestrings and also ones obtained by CBOR-serialising `Data`
+objects. However this will only test what we want when the lazy bytestring is
+encoded non-canonically, which in fact happens quite rarely. To make sure that
+we really do test some non-canonical inputs there are a few handwritten examples
+as well. -}
+test_nonCanonicalByteStringDecoding :: TestTree
+test_nonCanonicalByteStringDecoding =
+  let target = "This is a test." :: BS.ByteString
+
+      ch :: Char -> Word8
+      ch = fromIntegral . ord
+
+      input1 = BS.pack [ 0x01
+                       , 0x01, ch 'T'
```

Comment:
I assume 0x01 is the separator of the chunks, but can you add a comment pointing to where it comes from - e.g., a spec?

Reply:
> I assume 0x01 is the separator of the chunks

That's the tag for a bytestring: it said that in one of the earlier tests, but I've repeated it here and added a reference to the Plutus  Core specification. The other 0x01 bytes here are the starts of chunks of length 1.